Wikipedia/UI-V5/WMFLocationManager.m

```diff
@@ -68,14 +91,56 @@ - (void)startLocationUpdates {
 }
 
 - (void)startHeadingUpdates {
+    if (![[UIDevice currentDevice] isGeneratingDeviceOrientationNotifications]) {
+        [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+    }
+    @weakify(self);
+    self.orientationNotificationToken =
+        [[NSNotificationCenter defaultCenter] addObserverForName:UIDeviceOrientationDidChangeNotification
+                                                          object:nil
+                                                           queue:nil
+                                                      usingBlock:^(NSNotification* note) {
+        @strongify(self);
+        [self updateHeadingOrientation];
+    }];
+    [self updateHeadingOrientation];
     [self.locationManager startUpdatingHeading];
 }
 
+- (void)updateHeadingOrientation {
```

Comment:
yep, removed that, so now compass always points where it's supposed to whether we're face down/up, portrait/landscape, etc.